nifpga/bitfile.py

```diff
@@ -261,3 +677,13 @@ def number(self):
     def name(self):
         """ Returns the name of the FIFO. """
         return self._name
+
+    @property
+    def type(self):
+        return self._type
+
+    def is_fxp(self):
+        return isinstance(self._type, _FXP)
+
+    def is_composite(self):
+        return isinstance(self._type, _Cluster) or isinstance(self._type, _Array)
```

Comment:
Are FXP Fifos not also compsite? Since they are essentially just an array?

Reply:
FXP FIFOs are going to be the oddball out since they are always 64bits regardless of the actual size.

If you made a FIFO with a Cluster{FXP:32bit}, the FIFOs transfer size would be only 32bits.
If you made a FIFO with just a FXP:32bit value, the FIFOs transfer size will be 64bits, even though the last 32bits are unused.  So we have to handle them differently.

Can't change the past though...

Reply:
Thats interesting!
 <<TODO>> Time machine